File: 0429_med_hot_n-ary-tree-level-order-traversal/solution.py
"""
# Definition for a Node.
"""
from collections import deque
from typing import List, Optional


class Node:
    def __init__(self, val: Optional[int] = None, children: Optional[List['Node']] = None):
        self.val = val
        self.children = children


class Solution:
    def levelOrder(self, root: 'Node') -> List[List[int]]:
        if not root:
            return []
        
        result = []
        queue = deque()

        queue.append(root)

        while queue:
            level_result = []
            size = len(queue)
            for _ in range(size):
                node = queue.popleft()
                level_result.append(node.val)
                if node.children:
                    queue.extend(node.children)
            result.append(level_result)
            
        return result

File: 0429_med_hot_n-ary-tree-level-order-traversal/solution.java
/*
// Definition for a Node.
*/

import java.util.*;

class Node {
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val) {
        val = _val;
    }

    public Node(int _val, List<Node> _children) {
        val = _val;
        children = _children;
    }
};

class Solution {
    public List<List<Integer>> levelOrder(Node root) {
        if(root == null) {
            return new ArrayList<>();
        }

        List<List<Integer>> result = new ArrayList<>();

        Queue<Node> queue = new LinkedList<>();

        queue.offer(root);

        while (!queue.isEmpty()) {
            int size =  queue.size();
            List<Integer> levelResult = new ArrayList<>();
            while (size-- > 0) {
                Node node = queue.poll();
                levelResult.add(node.val);
                if(node.children != null) {
                    queue.addAll(node.children);
                }
            }
            result.add(levelResult);
        }

        return result;
    }
}